agent/lib/kontena/rpc/docker_container_api.rb

```diff
@@ -109,18 +109,72 @@ def top(id)
       def exec(id, cmd, opts = {})
         container = Docker::Container.get(id)
         container.exec(cmd, opts)
-      rescue Docker::Error::NotFoundError => exc
+      rescue Docker::Error::NotFoundError
         raise RpcServer::Error.new(404, 'Not found')
       end
 
+      def create_exec(id)
+        container = Docker::Container.get(id)
+        executor = Kontena::Actors::ContainerExec.new(container)
+        actor_id = "container_exec_#{executor.uuid}"
+        Celluloid::Actor[actor_id] = executor
+        { id: executor.uuid }
+      rescue Docker::Error::NotFoundError
+        raise RpcServer::Error.new(404, 'Container not found')
+      end
+
+      def run_exec(id, cmd)
+        actor_id = "container_exec_#{id}"
+        executor = Celluloid::Actor[actor_id]
+        if executor
+          executor.async.run(cmd)
```

Comment:
Not sure, should we return exec errors using exec session id? If we remove async here it will block until cmd is completed -> we should then also call this api with request (not notify).